### PR TITLE
[d] remove whitespaces at the end of lines of .cpp, .h and .lua files

### PR DESCRIPTION
**Pull request by HybridDog**
there's no other change, see https://github.com/minetest/minetest/pull/2737/files?w=1

### CODE REVIEW

File: src/json/jsoncpp.cpp [hunk 1]
```diff
@@ -6,28 +6,28 @@
 // //////////////////////////////////////////////////////////////////////
```
Comment (est31) [inline src/json/jsoncpp.cpp]: This is holy code one MUST NOT TOUCH.

Comment (HybridDog) [inline src/json/jsoncpp.cpp]: sorry, it did it automatically

**Comment by est31** [comment]
I think what @kwolekr said is right. Whitespace at the end of lines is bad, but one big change only creates merge conflicts. I will accept a PR though that only affects files for which no PRs (that don't need rebase) are open.

**Comment by HybridDog** [comment]
there are more than 90 pull requests

**Comment by rubenwardy** [comment]
Code style commits like this will make merge conflicts. Whenever I make a PR, my editor cleans up the file I'm in.

**Comment by HybridDog** [comment]
Some people don't use editors doing this.

**Comment by kilbith** [comment]
Please close this PR. There are too much files affected and the whitespace cleanup can be done progressively.

**Pull request closed without merging**